daemon/monitor.go

```diff
@@ -39,7 +39,8 @@ func (daemon *Daemon) handleContainerExit(c *container.Container, e *libcontaine
 
 	tsk, ok := c.Task()
 	if ok {
-		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		deleteTimeout := time.Duration(c.DeleteTimeout())
```

Comment:
time.Duration represents nanosecond count, so we need to either multiply by `time.Second` here or in `container.DeleteTimeout`.

```suggestion
		deleteTimeout := time.Duration(c.DeleteTimeout()) * time.Second
```

Reply:
Oof, that would explain the test failures, thanks.

Reply:
This also makes it apparent that the timeout value of 0 seconds gives `tsk.Delete` too little time to do anything meaningful.

I think we should also require the timeout to be greater than 0.

We can validate the config when creating the container:
https://github.com/moby/moby/blob/82d8f8d6e6dbc88ed437b8bf6c38399b46ba7d93/daemon/container.go#L245